javascript/javascript2/week1/hyf-bay/main.js

```diff
@@ -0,0 +1,133 @@
+        rating.setAttribute("class", "rating");
+        rating.innerHTML = product.rating;
+        listItem.appendChild(rating);
+        //ships-to
+        let shipping = document.createElement("div");
+        shipping.setAttribute("class", "shipping");
+        shipping.innerHTML = product.shipsTo;
+        listItem.appendChild(shipping);
+        //nested ul for shipping
+        let ShippingList = document.createElement("ul");
+        shipping.appendChild(ShippingList);
+        let shippingItemDk = document.createElement("li");
+        shippingItemDk.innerHTML = "Denmark";
+        ShippingList.appendChild(shippingItemDk);
+        let shippingItemSw = document.createElement("li");
+        shippingItemSw.innerHTML = "Sweden";
+        ShippingList.appendChild(shippingItemSw);
+        //button
+        let btn = document.createElement("button");
+        btn.innerHTML = "Add to cart";
+        listItem.appendChild(btn);
+        btn.onclick = () => addCart(product);
+        //appending li to ul
+        list.appendChild(listItem);
+    })
+}
+renderProducts(products);
+
+//filter/get products
+let arrayOfinputs = document.getElementsByTagName("input");
+console.log(arrayOfinputs);
+arrayOfinputs[0].addEventListener("input", function () {
+    let getProduct = arrayOfinputs[0].value;
+    console.log(getProduct);
+
+    //searching for the product
+    let searchProduct = products.filter((product) => {
+        return product.name.toLowerCase().includes(getProduct.toLowerCase());
+    })
+    console.log(searchProduct);
+    renderProducts(searchProduct);
+})
+
+//Showing products that ships to country
+let countrySelect = document.querySelector(".country select");
+console.log(countrySelect);
+countrySelect.addEventListener("change", function () {
+    let selectedCountry = countrySelect.value;
+    console.log(selectedCountry.toUpperCase());
+    let selectByCountry = products.filter((product) => {
+        for (let i = 0; i < product.shipsTo.length; i++) {
+            return product.shipsTo[i].toUpperCase().includes(selectedCountry.toUpperCase());
+        }
+    });
+
+    renderProducts(selectByCountry);
+})
+
+// Sort the products
+let sortSelect = document.querySelector(".sort select");
+console.log(sortSelect);
+sortSelect.addEventListener("change", function () {
+    let selectSortList = sortSelect.value;
+    console.log(selectSortList);
+    if (selectSortList === "cheap") {
+        let cheapFirstList = products.sort((product1, product2) => product1.price - product2.price);
+        renderProducts(cheapFirstList);
+
+    } else if (selectSortList === "expensive") {
+        let expensiveFirstList = products.sort((product1, product2) => product2.price - product1.price);
+        renderProducts(expensiveFirstList);
+    }
+})
+
+//Add cart
+let result = 0;
+function addCart(product) {
+    let selectCart = document.querySelector(".cart");
+    let selectUl = document.querySelector(".cart ul");
```

Comment:
Consider moving this variable outside the function because each call to `querySelector` is slow. If you remember the variable outside the function then you only need to call `querySelector` once.